eng/common/pipelines/templates/steps/create-apireview.yml

```diff
@@ -1,37 +1,73 @@
 parameters:
-  ArtifactPath: $(Build.ArtifactStagingDirectory)
-  Artifacts: []
-  ConfigFileDir: $(Build.ArtifactStagingDirectory)/PackageInfo
-  MarkPackageAsShipped: false
-  GenerateApiReviewForManualOnly: false
-  ArtifactName: 'packages'
-  PackageName: ''
-  SourceRootPath: $(Build.SourcesDirectory)
+  - name: ArtifactPath
+    type: string
+    default: $(Build.ArtifactStagingDirectory)
+  - name: Artifacts
+    type: object
+    default: []
+  - name: ConfigFileDir
+    type: string
+    default: $(Build.ArtifactStagingDirectory)/PackageInfo
+  - name: MarkPackageAsShipped
+    type: boolean
+    default: false
+  - name: GenerateApiReviewForManualOnly
+    type: boolean
+    default: false
+  - name: ArtifactName
+    type: string
+    default: 'packages'
+  - name: PackageName
+    type: string
+    default: ''
+  - name: SourceRootPath
+    type: string
+    default: $(Build.SourcesDirectory)
+  - name: PackageInfoFiles
+    type: object
+    default: []
 
 steps:
-  # ideally this should be done as initial step of a job in caller template
-  # We can remove this step later once it is added in caller
-  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
-    parameters:
-      WorkingDirectory: ${{ parameters.SourceRootPath }}
-
   # Automatic API review is generated for a package when pipeline runs irrespective of how pipeline gets triggered.
-  # Below condition ensures that API review is generated only for manual pipeline runs when flag GenerateApiReviewForManualOnly is set to true. 
+  # Below condition ensures that API review is generated only for manual pipeline runs when flag GenerateApiReviewForManualOnly is set to true.
   - ${{ if or(ne(parameters.GenerateApiReviewForManualOnly, true), eq(variables['Build.Reason'], 'Manual')) }}:
+    # ideally this should be done as initial step of a job in caller template
+    # We can remove this step later once it is added in caller
+    - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+      parameters:
+        WorkingDirectory: ${{ parameters.SourceRootPath }}
+
+    - pwsh: |
+        # If the Artifacts array is set create the list PackageInfo files from them otherwise use
+        # the PackageInfoFiles that were passed in to create the list.
```

Comment:
I assume there shouldn't be a case where both are set and we should aggregate them?

Reply:
No. It's an either or. Other languages will continue to use Artifacts and, as far as I know, Java is the only one that will need to use the PackageInfo list.